### ISSUE TITLE
zzz sometimes causes frozen desktop after wakeup

### ISSUE THREAD

**Issue by notthewave** [void-linux/void-packages#33813]
### System

* xuname:  
  *Void 5.13.19_1 x86_64-musl GenuineIntel uptodate rDFFFFFFFFFFFF*
* package:  
  *affected package(s) including the version*: ``pm-utils-1.4.1_8``

### Expected behavior
does not freeze after wakeup

### Actual behavior
sometimes freezes after wakeup

### Steps to reproduce the behavior
1. run zzz and wait for a few minutes

### Aditional Info
probably does not happen to everyone, so I suppose this is caused by something else I have installed.

here are my running services:
```
NetworkManager	agetty-tty2  agetty-tty5   bluetoothd	dbus	  mdnsd  seatd  ufw
acpid		agetty-tty3  agetty-tty6   cups-browsed  dcron	  nginx  tlp	virtlockd
agetty-tty1	agetty-tty4  avahi-daemon  cupsd	libvirtd  ntpd   udevd  virtlogd
```

**tested with tty:**
recognizes keyboard input but can't stop zzz (CTRL+C)

**Comment by Animeshz** [comment]
Do you have Ryzen mobile APU? I guess its a [kernel (upstream) bug](https://gitlab.freedesktop.org/drm/amd/-/issues/1721), which still isn't resolved (or isn't figured out what causing it, otherwise). What is kernel version do you have (`xbps-query -l | awk '{ print $2 }' | grep linux5.`)?

**Comment by notthewave** [comment]
I'm on Intel (x230t thinkpad)
kernel version 5.13.19_1

**Comment by Animeshz** [comment]
Well, then I think it might be really some issue with the void configuration I think?

 Cuz for a lot of people that issue was fixed, whereas I'm still experiencing in latest kernel and firmware 🤔🤔

**Comment by Animeshz** [comment]
Hey,

- Does it happens with resume within 3s of suspend?
- Can you try exiting the wm/de using keybind and reenter and do the same, if I do it twice then the screen never comes back on (remains black), I have to hard power reset then. Is the behavior same?

If we face the same issue then its highly sure that there is something in the config itself which have some problem.

**Comment by foopub** [comment]
Check the logs from when it happens i.e. dmesg and svlogtail.
Does Ctrl + D work to stop zzz? 
By desktop I presume you mean some X environment, maybe check the Xorg logs too (I think in /var/log/).

Post anything suspicious you find.

**Comment by notthewave** [comment]
[    4.251928] ACPI Warning: SystemIO range 0x0000000000000428-0x000000000000042F conflicts with OpRegion 0x0000000000000400-0x000000000000047F (\_SB.PCI0.LPC.PMIO) (20210331/utaddress-204)

**Comment by notthewave** [comment]
I'm on wayland and use seatd but somehow I still need elogind installed in order for it to run. could be the cause...

**Comment by notthewave** [comment]
jup elogind seems to cause it

**Comment by notthewave** [comment]
kernel seems to be the issue tried previous version and it works as expected

**Comment by notthewave** [comment]
believe I found the devil! 
had `tpacpi-bat` installed

**Comment by notthewave** [comment]
lol nope

**Comment by Animeshz** [comment]
@foopub I've posted the dmesg logs in the mentioned issue over amd (my first comment) there's an amdgpu issue as it seems, other than that the Xorg logs are here 
[Xorg.0.log](https://github.com/void-linux/void-packages/files/7522314/Xorg.0.log) and nothing seem suspicious.
Ctrl+D does nothing.

**Comment by notthewave** [comment]
I disabled tlp and it seems to work now after closing and opening the lid aprox. 50 times.
https://github.com/linrunner/TLP/issues/593
I disabled it before but it did not have any effect iirc

**Comment by Animeshz** [comment]
I have got everything fixed, it was because AMD_PMC was not set in the kernel config. I ran configure, ran menuconfig, added it and compiled and installed and now everything works!

**Comment by Animeshz** [comment]
Not sure why that module is not enabled, it is responsible for s2idle (s0ix) states in AMD notebooks. Please enable it as module, custom kernel like [linux-tkg put it as a module](https://github.com/Frogging-Family/linux-tkg/blob/d339b16b7b2758534a99277b5b2a994656e2a2e1/linux-tkg-config/5.15/config.x86_64#L8349).

**Comment by leahneukirchen** [comment]
Fixed in dc1fd41283.
